### PR TITLE
[cli] Download automatically exposed System Environment Variables when running `vercel env pull`

### PR DESCRIPTION
**Pull request by lucleray**
Ref: https://app.clubhouse.io/vercel/story/15489

Download automatically exposed system env variables when running `vc env pull`:
```
VERCEL="1"
VERCEL_ENV="development"
VERCEL_URL=""
VERCEL_GIT_PROVIDER=""
VERCEL_GIT_REPO_SLUG=""
VERCEL_GIT_REPO_OWNER=""
VERCEL_GIT_REPO_ID=""
VERCEL_GIT_COMMIT_REF=""
VERCEL_GIT_COMMIT_SHA=""
VERCEL_GIT_COMMIT_MESSAGE=""
VERCEL_GIT_COMMIT_AUTHOR_LOGIN=""
VERCEL_GIT_COMMIT_AUTHOR_NAME=""
```

Also adds support for references to `VERCEL_URL` system env when running `vc dev`. Previously, it would leave the value empty when running `vc dev`, this PR fixes it and correctly populates it.

Also fix exposing `VERCEL_REGION` and `NOW_REGION`. They should only be exposed at "runtime" to mirror production more closely.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [x] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/now-cli/test/integration.js [hunk 1]
```diff
@@ -539,7 +539,7 @@ test('Deploy `api-env` fixture and test `vercel env` command', async t => {
   async function nowEnvAddSystemEnv() {
     const now = execa(
       binaryPath,
-      ['env', 'add', 'system', 'VERCEL_URL', ...defaultArgs],
+      ['env', 'add', 'system', 'NEXT_PUBLIC_VERCEL_URL', ...defaultArgs],
```
Comment (lucleray) [inline packages/now-cli/test/integration.js:542]: Updated this test to properly test that references to system env variables are functioning when the env name is different than the "system value" (with `vc`, `vc dev` and `vc env pull`).

**Review by AndyBitz — DISMISSED** [review]

**Comment by AndyBitz** [comment]
Approved, but unit tests have to be fixed first

**Review by AndyBitz — DISMISSED** [review]

**Review by styfle — DISMISSED** [review]

File: packages/now-cli/src/commands/dev/dev.ts [hunk 2]
```diff
@@ -101,16 +99,9 @@ export default async function dev(
...
-        client,
-        project,
-        ProjectEnvTarget.Development
-      ),
-      project.autoExposeSystemEnvs
-        ? exposeSystemEnvs(output, client, project.id)
-        : undefined,
+    [{ envs: projectEnvs }, { systemEnvValues }] = await Promise.all([
+      getDecryptedEnvRecords(output, client, project.id),
+      getSystemEnvValues(output, client, project.id),
```
Comment (nkzawa) [inline packages/now-cli/src/commands/dev/dev.ts]: Looks like the value is never used when `project.autoExposeSystemEnvs` is `false` so it'd be better to not retrieve values in that case.

File: packages/now-cli/src/commands/env/pull.ts [hunk 3]
```diff
@@ -84,15 +85,20 @@ export default async function pull(
...
-        project,
-        ProjectEnvTarget.Development
-      )
+  const [
+    { envs: projectEnvs },
+    { systemEnvValues },
+  ] = await withSpinner('Downloading', () =>
+    Promise.all([
+      getDecryptedEnvRecords(output, client, project.id),
+      getSystemEnvValues(output, client, project.id),
```
Comment (nkzawa) [inline packages/now-cli/src/commands/env/pull.ts]: Same with the above comment.

File: packages/now-cli/src/commands/dev/dev.ts (hunk 2 above)
Comment (lucleray) [reply packages/now-cli/src/commands/dev/dev.ts]: oh great point, fixed both in https://github.com/vercel/vercel/pull/5451/commits/15b7b9e52eca99a4fcc6751f52a1a44fe2a35da5 👍

**Review by styfle — APPROVED** [review]

**Review by nkzawa — APPROVED** [review]

**Pull request merged**